packages/devtools_app/lib/src/debugger/file_search.dart

```diff
@@ -18,52 +18,73 @@ const double autocompleteMatchTileHeight = 50.0;
 
 class FileSearchField extends StatefulWidget {
   const FileSearchField({
-    @required this.controller,
+    @required this.debuggerController,
+    @required this.autoCompleteController,
   });
 
-  final DebuggerController controller;
+  final DebuggerController debuggerController;
+  final AutoCompleteController autoCompleteController;
 
   @override
   _FileSearchFieldState createState() => _FileSearchFieldState();
 }
 
 class _FileSearchFieldState extends State<FileSearchField>
     with SearchFieldMixin, AutoDisposeMixin {
-  AutoCompleteController _autoCompleteController;
-
   final _scriptsCache = <String, ScriptRef>{};
 
   final fileSearchFieldKey = GlobalKey(debugLabel: 'fileSearchFieldKey');
 
+  String _query;
+  List<ScriptRef> _matches;
+
   @override
   void initState() {
     super.initState();
 
-    _autoCompleteController = AutoCompleteController()..currentDefaultIndex = 0;
+    widget.autoCompleteController.currentDefaultIndex = 0;
 
     addAutoDisposeListener(
-        _autoCompleteController.searchNotifier, _handleSearch);
-    addAutoDisposeListener(_autoCompleteController.searchAutoCompleteNotifier,
+        widget.autoCompleteController.searchNotifier, _handleSearch);
+    addAutoDisposeListener(
+        widget.autoCompleteController.searchAutoCompleteNotifier,
         _handleAutoCompleteOverlay);
 
+    _query = widget.autoCompleteController.search;
+    _matches = widget.debuggerController.sortedScripts.value;
+
     // Open the autocomplete results immediately before a query is entered:
     SchedulerBinding.instance.addPostFrameCallback((_) => _handleSearch());
   }
 
   void _handleSearch() {
-    final query = _autoCompleteController.search;
-    final matches = findMatches(query, widget.controller.sortedScripts.value);
+    final previousQuery = _query;
+    final currentQuery = widget.autoCompleteController.search;
+
+    // If the current query is a continuation of the previous query, then
+    // whittle down the matches. Otherwise search through all scripts:
+    final scripts = currentQuery.startsWith(previousQuery)
+        ? _matches
+        : widget.debuggerController.sortedScripts.value;
```

Comment:
the line widget.autocompleteController occurs a lot now. Feel free to add an
autocompleteController getter to the state class that just returns widget.autocompleteController to make this code a little less verbose.

Reply:
Back to creating it in the file search state, so it's no longer on the widget